internal/cache/util/util_test.go

```diff
@@ -136,18 +149,19 @@ func (ut *utilTest) Test_CreateFile_FilePerm0755() {
 
 	file, err := CreateFile(ut.fileSpec, ut.flag)
 
-	ut.assertFileAndDirCreation(file, err)
+	ut.assertFileAndDirCreationWithGivenDirMode(file, err, 0755)
 	ExpectEq(nil, file.Close())
 }
 
-func (ut *utilTest) Test_CreateFile_FilePerm0544() {
-	ut.fileSpec.Perm = os.FileMode(0544)
-
-	file, err := CreateFile(ut.fileSpec, ut.flag)
-
-	ut.assertFileAndDirCreation(file, err)
-	ExpectEq(nil, file.Close())
-}
+// TODO: should we remove this test as we use same perms for file and dir and dir needs min 7 perm as owner to be able to create file
```

Comment:
yes we can remove this. Since the function now creates both file and dirs with same permissions. we need to adjust our tests accordingly, so you can write tests (if not already present) 
(a) if directory is already present then different file perms including <700.
(b) if directory is not already present then couple of cases creating dir+files for perm >=700 
(c) if directory is not already present then couple of cases trying to create dir+files for perm <700 should fail.

Reply:
yes we can remove this. Since the function now creates both file and dirs with same permissions. we need to adjust our tests accordingly, so you can write tests (if not already present) 
(a) if directory is already present then different file perms including <700. _[already present - Test_CreateFile_FilePresentWithLessAccess]_
(b) if directory is not already present then couple of cases creating dir+files for perm >=700 _[multiple scenarios are already present for this]_
(c) if directory is not already present then couple of cases trying to create dir+files for perm <700 should fail. _[modified this commented test for this usecase]_